src/SFA.DAS.Forecasting.Application.UnitTests/AccountProjection/Queries/WhenGettingProjectionSummaryForAnAccount.cs

```diff
@@ -0,0 +1,121 @@
+            _cancellationToken = new CancellationToken();
+            _service = new Mock<IAccountProjectionService>();
+            var accountProjectionSummary = new AccountProjectionSummary(
+                ExpectedAccountId, 
+                DateTime.Today,
+                12,
+                123.56M,
+                654.32M);
+            _service.Setup(x => x.GetProjectionSummary(ExpectedAccountId, DateTime.Today, 12)).ReturnsAsync(accountProjectionSummary);
+
+            _handler = new GetAccountProjectionSummaryQueryHandler(_validator.Object, _service.Object);
+        }
+
+        [Test]
+        public async Task Then_The_Query_Is_Validated()
+        {
+            //Act
+            await _handler.Handle(_query, _cancellationToken);
+
+            //Assert
+            _validator.Verify(x => x.ValidateAsync(_query), Times.Once);
+
+        }
+
+        [Test]
+        public void Then_If_The_Query_Fails_Validation_Then_An_Argument_Exception_Is_Thrown()
+        {
+            //Arrange
+            _validator.Setup(x => x.ValidateAsync(It.IsAny<GetAccountProjectionSummaryQuery>()))
+                .ReturnsAsync(new ValidationResult { ValidationDictionary = new Dictionary<string, string> { { "", "" } } });
+
+            //Act Assert
+            Assert.ThrowsAsync<ArgumentException>(async () => await _handler.Handle(_query, _cancellationToken));
+        }
+
+        [Test]
+        public async Task Then_The_Return_Type_Is_Assigned_To_The_Response()
+        {
+            //Act
+            var actual = await _handler.Handle(_query, _cancellationToken);
+
+            //Assert
+            Assert.IsAssignableFrom<GetAccountProjectionSummaryResult>(actual);
+        }
+
+        [Test]
+        public async Task Then_The_Repository_Is_Called_With_The_Request_Details()
+        {
+            //Act
+            await _handler.Handle(_query, _cancellationToken);
+
+            //Assert
+            _service.Verify(x => x.GetProjectionSummary(ExpectedAccountId, DateTime.Today, NumberOfMonths));
+        }
+
+        [Test]
+        public async Task Then_The_Values_Are_Returned_In_The_Response()
+        {
+            //Arrange
+            decimal expectedFundsIn = 100.00M;
+            decimal expectedFundsOut = 999.99M;
+            int expectedNumberOfmonths = 6;
+            var startDate = DateTime.Today;
+            var accountProjectionSummary = new AccountProjectionSummary(ExpectedAccountId, startDate, expectedNumberOfmonths, expectedFundsIn, expectedFundsOut);
+            _service.Setup(x => x.GetProjectionSummary(ExpectedAccountId, DateTime.Today, expectedNumberOfmonths)).ReturnsAsync(accountProjectionSummary);
+            _query.NumberOfMonths = expectedNumberOfmonths;
+
+            //Act
+            var actual = await _handler.Handle(_query, _cancellationToken);
+
+            //Assert
+            Assert.AreEqual(ExpectedAccountId, actual.AccountId);
```

Comment:
Again we could use fluent assertions to do a single assert on an expected object response vs the actual

Reply:
I've written the asserts to match the existing tests which don't use FluentAssertions .